### PR TITLE
[SPARK-33908][CORE] Refactor SparkSubmitUtils.resolveMavenCoordinates() 's return parameter

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Per discuss in  https://github.com/apache/spark/pull/29966#discussion_r531917374
We'd better change `SparkSubmitUtils.resolveMavenCoordinates()` 's return value as `Seq[String]`

### Why are the changes needed?
refactor code

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existed UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
FYI @xkrogen @maropu @dongjoon-hyun

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 1]
```diff
@@ -1362,17 +1363,17 @@ private[spark] object SparkSubmitUtils {
...
   def resolveMavenCoordinates(
       coordinates: String,
       ivySettings: IvySettings,
       transitive: Boolean,
       exclusions: Seq[String] = Nil,
-      isTest: Boolean = false): String = {
+      isTest: Boolean = false): Seq[String] = {
     if (coordinates == null || coordinates.trim.isEmpty) {
-      ""
+      Seq.empty[String]
```
Comment (maropu) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala:1376]: nit: Nil?

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala [hunk 2]
```diff
@@ -137,7 +133,7 @@ class SparkSubmitUtilsSuite extends SparkFunSuite with BeforeAndAfterAll {
         SparkSubmitUtils.buildIvySettings(Option(repo), Some(tempIvyPath)),
         transitive = true,
         isTest = true)
-      assert(jarPath.indexOf(tempIvyPath) >= 0, "should use non-default ivy path")
+      assert(jarPath.forall(_.indexOf(tempIvyPath) >=0), "should use non-default ivy path")
```
Comment (maropu) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala:136]: nit: `>= 0`

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala [hunk 3]
```diff
@@ -176,9 +172,9 @@ class SparkSubmitUtilsSuite extends SparkFunSuite with BeforeAndAfterAll {
         SparkSubmitUtils.buildIvySettings(None, Some(tempIvyPath)),
         transitive = true,
         isTest = true)
-      assert(jarPath.indexOf("mylib") >= 0, "should find artifact")
-      assert(jarPath.indexOf(tempIvyPath) >= 0, "should be in new ivy path")
-      assert(jarPath.indexOf("mydep") >= 0, "should find dependency")
+      assert(jarPath.exists(_.indexOf("mylib") >=0), "should find artifact")
```
Comment (maropu) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala:175]: ditto

**Comment by maropu** [comment]
Looks fine otherwise if the tests pass.

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala (hunk 3 above)
Comment (AngersZhuuuu) [reply core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala:175]: Done

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala:136]: Done

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Done

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
retest this please

**Review by srowen — APPROVED** [review]

**Comment by AngersZhuuuu** [comment]
Gentle ping @maropu

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 4]
```diff
@@ -1201,7 +1202,7 @@ private[spark] object SparkSubmitUtils {
    */
```
Comment (xkrogen) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala:1202]: @AngersZhuuuu  FYI you missed updating the `@return` Scaladoc tag here, as well as the description which explicitly mentions a comma-delimited list. Would you mind submitting a follow-on to update?

**Review by xkrogen — COMMENTED** [review]
Many thanks for this change @AngersZhuuuu ! Looks much cleaner this way.

**Pull request closed without merging**